packages/commcare/test/index.js

```diff
@@ -679,11 +678,131 @@ describe('HTTP wrappers', () => {
       },
     };
 
-    const { data, response } = await 
-      request('GET', '/a/asri/api/v0.5/case')
-    (state);
+    const { data, response } = await request(
+      'GET',
+      '/a/asri/api/v0.5/case'
+    )(state);
 
     expect(data).to.haveOwnProperty('case_id');
     expect(response.statusCode).to.equal(200);
   });
-});
\ No newline at end of file
+});
+
+describe('Bulk', () => {
+  it('should submit JSON as an xls file', async () => {
+    let formdata;
+
+    testServer
+      .intercept({
+        path: `/a/${domain}/importer/excel/bulk_upload_api/`,
+        method: 'POST',
+      })
+      .reply(200, req => {
+        // save out the form data that was upladed
+        formdata = req.body;
+
+        // simulate a return from commcare
+        return { code: 200, message: 'success' };
+      });
+
+    const state = {
+      configuration: {
+        hostUrl,
+        domain,
+        appId: app,
+        username: 'user',
+        password: 'password',
+      },
+    };
+
+    const { data } = await execute(
+      bulk('case-data', [{ name: 'Mamadou', phone: '000000' }], {
+        case_type: 'student',
+        search_field: 'external_id',
+        create_new_cases: 'on',
+      })
+    )(state);
+
+    // The response  should be on state.data
+    expect(data.code).to.equal(200);
+    expect(data.message).to.equal('success');
+
+    // And the adaptor should have uploaded a reasonable looking formdata object
+    expect(formdata.get('case_type')).to.not.be.undefined;
+    expect(formdata.get('case_type')).to.equal('student');
+    expect(formdata.get('search_field')).to.equal('external_id');
+    expect(formdata.get('create_new_cases')).to.equal('on');
+  });
+  it('should successfully submit a lookup-table file', async () => {
```

Comment:
Just being picky, your not really submitting a file, the file gets created by the adaptor behind the scene. So may be `should successfully bulk upload lookup-table data`